### ISSUE TITLE
SpanExt name collision with proc_macro2

### ISSUE THREAD

**Issue by bart358** [Juici/int-enum-rs#13]
`SpanExt` defines a method:
```rust
fn end(&self) -> Span;
```
And implements it on `Span`. However, `proc-marco2` with feature `span-locations` also defines a method on `Span`:
```rust
    #[cfg(span_locations)]
    #[cfg_attr(doc_cfg, doc(cfg(feature = "span-locations")))]
    pub fn end(&self) -> LineColumn {
        self.inner.end()
    }
```
When calling `end` on a `Span` object with that feature enabled the above method takes precedence. This breaks code in an `ast` module where it defines:
```rust
diag = diag.span_error(v.span().end(), "missing discriminant");
```
Since the `SpanExt` is in scope the intention was for `.end()` to return a `Span` object which works since it implements `MultiSpan` required by `span_error`. With `span-locations` enabled It returns a `LineColumn` which does not implement `MultiSpan`:
```rust
   --> int-enum-1.0.0/src/ast.rs:120:36
    |
120 |             diag = diag.span_error(v.span().end(), "missing discriminant");
    |                         ---------- ^^^^^^^^^^^^^^ the trait `proc_macro2_diagnostics::diagnostic::MultiSpan` is not implemented for `LineColumn`
    |                         |
    |                         required by a bound introduced by this call
    |
    = help: the following other types implement trait `proc_macro2_diagnostics::diagnostic::MultiSpan`:
              proc_macro2::Span
              Vec<proc_macro2::Span>
              &'a [proc_macro2::Span]
note: required by a bound in `proc_macro2_diagnostics::Diagnostic::span_error`
``` 
The solution would be to refer to the trait explicitly:
```rust
diag = diag.span_error(<Span as SpanExt>::end(&v.span()), "missing discriminant");
```

**Comment by Juici** [comment]
I just noticed the build errors, I'll fix it.

**Comment by Juici** [comment]
This should be fixed by #15 for version 1.0.1.
